doc/_extensions/zephyr/manifest_projects_table.py

```diff
@@ -0,0 +1,142 @@
+"""
+Manifest Revisions Table
+========================
+
+This extension allows to render a table containing the revisions of the projects
+present in a manifest file.
+
+Usage
+*****
+
+This extension introduces a new directive: ``manifest-projects-table``. It can
+be used in the code as::
+
+    .. manifest-projects-table::
+        :filter: active
+
+where the ``:filter:`` option can have the following values: active, inactive, all.
+
+Options
+*******
+
+- ``manifest_projects_table_manifest``: Path to the manifest file.
+
+Copyright (c) Nordic Semiconductor ASA 2022
+Copyright (c) Intel Corp 2023
+SPDX-License-Identifier: Apache-2.0
+"""
+
+import re
+from typing import Any, Dict, List
+
+from docutils import nodes
+from docutils.parsers.rst import directives
+from sphinx.application import Sphinx
+from sphinx.util.docutils import SphinxDirective
+from west.manifest import Manifest
+
+
+__version__ = "0.1.0"
+
+
+class ManifestProjectsTable(SphinxDirective):
+    """Manifest revisions table."""
+
+    option_spec = {
+        "filter": directives.unchanged,
+    }
+
+    @staticmethod
+    def rev_url(base_url: str, rev: str) -> str:
+        """Return URL for a revision.
+
+        Notes:
+            Revision format is assumed to be a git hash or a tag. URL is
+            formatted assuming a GitHub base URL.
+
+        Args:
+            base_url: Base URL of the repository.
+            rev: Revision.
+
+        Returns:
+            URL for the revision.
+        """
+
+        if re.match(r"^[0-9a-f]{40}$", rev):
+            return f"{base_url}/commit/{rev}"
+
+        return f"{base_url}/releases/tag/{rev}"
+
+    def run(self) -> List[nodes.Element]:
+        active_filter = self.options.get("filter", None)
+
+        # sort manifest projects accounting
+        manifest = Manifest.from_file(self.env.config.manifest_projects_table_manifest)
+        projects = []
+        for project in manifest.projects:
```

Comment:
What are the odds of eventually adding more optional metadata to the Project? Would be nice for this table to include the link to the docs for the module if available.

Reply:
a more detailed description field was already added, we probably want to add more, I agree.